### PR TITLE
ch4: replace MPIDI_POSIX_RESIZE_TO_MAX_ALIGN

### PR DESCRIPTION
**Pull request by mjwilkins18**
## Pull Request Description
Fixes compilation bug, compilation failed with ofi and cray cc compiler.

## Author Checklist
* [x] **Provide Description** 
      Particularly focus on _why_, not _what_. Reference background, issues, test failures, xfail entries, etc.
* [x] **Commits Follow Good Practice**
      Commits are self-contained and do not do two things at once. 
      Commit message is of the form: `module: short description` 
      Commit message explains what's in the commit.
* [x] **Passes All Tests**
      Whitespace checker. Warnings test. Additional tests via comments.
* [x] **Contribution Agreement**
      For non-Argonne authors, check [contribution agreement](http://www.mpich.org/documentation/contributor-docs/). 
      If necessary, request an explicit comment from your companies PR approval manager.

### CODE REVIEW

**Comment by hzhou** [comment]
This is such a generic semantics I think we should move it to `src/mpl/mpl_base.h` and rename it to `MPL_ROUNDUP(alignment)`.

**Comment by mjwilkins18** [comment]
That sounds good to me. Should we make a separate pull request for that change? To me, fixing the compilation error and updating the semantic are separate.

**Comment by hzhou** [comment]
> 
> 
> That sounds good to me. Should we make a separate pull request for that change? To me, fixing the compilation error and updating the semantic are separate.

It is the same fix. One is like a bandaid. The other is like a surgery. IMO, there is no need to apply a bandaid if you are going to immediately follow with a surgery.

**Comment by mjwilkins18** [comment]
Ok, I will make more commits with the bigger change when ready

**Comment by mjwilkins18** [comment]
MPIDI_POSIX_RESIZE_TO_MAX_ALIGN has to changed to MPL_ROUNDUP as requested.

File: src/mpl/include/mpl_base.h [hunk 1]
```diff
@@ -159,5 +159,8 @@ typedef bool _Bool;
 
 #define MPL_ROUND_UP_ALIGN(a, alignment) (((a) + ((alignment) - 1)) & (~((alignment) - 1)))
 #define MPL_ROUND_DOWN_ALIGN(a, alignment) ((a) & (~((alignment) - 1)))
+#define MPL_ROUNDUP(alignment) \
```
Comment (hzhou) [inline src/mpl/include/mpl_base.h:162]: Oh, we already have `MPL_ROUND_UP_ALIGN`, just use it, there is no need for an extra macro.

**Review by hzhou — CHANGES_REQUESTED** [review]

File: src/mpl/include/mpl_base.h (hunk 1 above)
Comment (mjwilkins18) [reply src/mpl/include/mpl_base.h:162]: Done. A small note: the existing macro assumes you are aligning to a power of 2, while the new one does not. I do not think it is an issue for this use note, but the semantics of the two are not identical

File: src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h [hunk 2]
```diff
@@ -80,7 +80,7 @@ MPIDI_POSIX_eager_send(int grank, MPIDI_POSIX_am_header_t * msg_hdr, const void
      * tail. */
     cell->payload_size = 0;
     if (am_hdr) {
-        MPI_Aint resized_am_hdr_sz = MPIDI_POSIX_RESIZE_TO_MAX_ALIGN(am_hdr_sz);
+        MPI_Aint resized_am_hdr_sz = MPL_ROUNDUP(am_hdr_sz);
```
Comment (hzhou) [inline src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h:83]: What I meant is simply use this:
```
MPI_Aint resized_am_hdr_sz = MPL_ROUND_UP_ALIGN(am_hdr_sz, MAX_ALIGNMENT);
```

No need to add additional macro in `mpl_base.h`.

**Review by hzhou — CHANGES_REQUESTED** [review]

File: src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h (hunk 2 above)
Comment (hzhou) [reply src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h:83]: Could you also squash into a single commit? Thanks.

Comment (mjwilkins18) [reply src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h:83]: I understand, and I made the change but the commits are not showing here, but I can see them on the branch for this PR. Sorry for the confusion, working on a fix

Comment (hzhou) [reply src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h:83]: I see. Are you familiar with git interactive rebasing? https://hackernoon.com/beginners-guide-to-interactive-rebasing-346a3f9c3a6d

Comment (mjwilkins18) [reply src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h]: Yep, done

File: src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h [hunk 3]
```diff
@@ -80,7 +80,7 @@ MPIDI_POSIX_eager_send(int grank, MPIDI_POSIX_am_header_t * msg_hdr, const void
      * tail. */
     cell->payload_size = 0;
     if (am_hdr) {
-        MPI_Aint resized_am_hdr_sz = MPIDI_POSIX_RESIZE_TO_MAX_ALIGN(am_hdr_sz);
+        MPI_Aint resized_am_hdr_sz = MPL_ROUND_UP_ALIGN(am_hdr_sz, MAX_ALIGNMENT);
```
Comment (hzhou) [inline src/mpid/ch4/shm/posix/eager/iqueue/iqueue_send.h:83]: Let's still delete `MPIDI_POSIX_RESIZE_TO_MAX_ALIGN` from `posix_am.h`.

**Review by hzhou — CHANGES_REQUESTED** [review]

**Comment by hzhou** [comment]
test:mpich/ch4/most

**Review by hzhou — APPROVED** [review]
Thanks for the changes and fix!

**Pull request merged**